plugins/cratedb/database_credentials.go

```diff
@@ -0,0 +1,82 @@
+package cratedb
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func DatabaseCredentials() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.DatabaseCredentials,
+		DocsURL:       sdk.URL("https://crate.io/docs/crate/crash/en/latest/run.html#environment-variables"),
+		ManagementURL: sdk.URL("https://console.cratedb.cloud/account/settings"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Host,
+				MarkdownDescription: "CrateDB host to connect to.",
+				Optional:            true,
+			},
+			{
+				Name:                fieldname.Port,
+				MarkdownDescription: "Port used to connect to CrateDB.",
+				Optional:            true,
+			},
+			{
+				Name:                fieldname.User,
+				MarkdownDescription: "CrateDB user to authenticate as.",
+				Optional:            true,
+			},
+			{
+				Name:                fieldname.Password,
+				MarkdownDescription: "Password used to authenticate to CrateDB.",
+				Secret:              true,
+			},
+			{
+				Name:                fieldname.Database,
+				MarkdownDescription: "Database name to connect to.",
+				Optional:            true,
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
```

Comment:
The env var provisioner is great here, but it only provisions the password. The user/host/port end up not being used at all. So if someone actually sets these fields in their 1Password item, those will be ignored.

Let's specify those fields as arguments in the command line being run. 

To do this we need to take a few steps:

1. Create a new `CrashProvisioner` that implements the `Provisioner` interface. 
2. In the `Provision` function first call the env var provisioner to provision the password as an env var
3. Modify `ProvisionOutput.CommandLine` directly by adding the required flags. E.g. `--username <username_stored_in_1P>`. `<username_stored_in_1P>`  is accessible via `ProvisionInput.ItemField[fieldname.User]`.

Let me know if any questions

Reply:
any reference which can help me??
As I am new to Go

Reply:
Here's one example from the redis cli contribution. It's a discussion over the changes that are relevant to you: https://github.com/1Password/shell-plugins/pull/276#discussion_r1244843892

Reply:
I have just tried.

please check it

Reply:
Looks good